torch/utils/data/datapipes/_typing.py

```diff
@@ -375,6 +407,15 @@ def wrap_generator(*args, **kwargs):
                         response = gen.send(request)
             except StopIteration as e:
                 return e.value
+            except Exception as e:
+                # TODO: Simplify the traceback message to skip over `response = gen.send(None)`
+                #       Part of https://github.com/pytorch/data/issues/284
+                datapipe = args[0]
+                msg = "thrown by __iter__ of"
+                full_msg = f"{msg} {datapipe.__class__.__name__}({_generate_input_args_string(datapipe)})"
```

Comment:
When the Error comes from singleton iterator. This part of message becomes duplicated.
You can run the following example:
```py
source_dp = IterableWrapper(range(10))
it1 = iter(source_dp)
next(it1)
it2 = iter(source_dp)
next(it2)
next(it1)
```
The Error message is:
```py
RuntimeError: This iterator has been invalidated because another iterator has been created from the same IterDataPipe: IterableWrapperIterDataPipe(deepcopy=True, iterable=range(0, 10))
This may be caused multiple references to the same IterDataPipe. We recommend using `.fork()` if that is necessary.
For feedback regarding this single iterator per IterDataPipe constraint, feel free to comment on this issue: https://github.com/pytorch/data/issues/45.
This exception is thrown by __iter__ of IterableWrapperIterDataPipe(deepcopy=True, iterable=range(0, 10))
```

cc: @NivekT

Reply:
You are saying that we don't want `IterableWrapperIterDataPipe(deepcopy=True, iterable=range(0, 10))` to appear twice?

Reply:
I am open to that idea, but it looks fine to me.

Reply:
That would be the ideal case, IMO. We use `This exception is thrown by __iter__ of IterableWrapperIterDataPipe(deepcopy=True, iterable=range(0, 10))` to indicate the Error comes from internal logic of `__iter__`. But, the actual Error comes from the duplicate iterators. I think this message is a little misleading.

Reply:
I see your point, I can fix that.

Reply:
Thank you. Not high-pri though.